src/Microsoft.Health.Fhir.Core/Features/Search/SearchService.cs

```diff
@@ -51,8 +51,16 @@ protected SearchService(ISearchOptionsFactory searchOptionsFactory, IFhirDataSto
         {
             SearchOptions searchOptions = _searchOptionsFactory.Create(resourceType, queryParameters);
 
-            // Execute the actual search.
-            return await SearchInternalAsync(searchOptions, cancellationToken);
+            try
+            {
+                // Execute the actual search.
+                return await SearchInternalAsync(searchOptions, cancellationToken);
+            }
+            catch (Exception)
```

Comment:
This might be a problem since it treats any error as invalid continuation token. Instead, I think we should create our own InvalidContinuationTokenException or something like that and catch specific exception type in `FhirCosmosSearchService` and convert the exception type.

Reply:
This can be updated in https://github.com/microsoft/fhir-server/pull/589

Reply:
The specific exception type is a DocumentClientException. The issue I had when I tried to catch it specifically is that the class definition can't be found. It's in the Azure namespace so I don't know why a reference can't be made to it.